clients/src/main/java/org/apache/kafka/common/protocol/Errors.java

```diff
@@ -87,6 +89,13 @@
  * Do not add exceptions that occur only on the client or only on the server here.
  */
 public enum Errors {
+    UNKNOWN_RETRIABLE(-2, "The client received an unexpected error code when processing the response",
```

Comment:
Could we adjust the message a bit so that UNKNOWN_RETRIABLE can be distinguished from UNKNOWN? It would also be useful to add a comment on how it should be used differently from UNKNOWN.